File: examples/python/best-practices/avoid-individual-operations.py
@durable_execution
def lambda_handler(event: dict, context: DurableContext) -> list:
    results = []
    # Creating a step for each item - too many checkpoints!
    for i, item in enumerate(event["items"]):
        result = context.step(
            lambda _, item=item: process_item(item),
            name=f"process_item_{i}"
        )
        results.append(result)
    return results

File: examples/java/best-practices/avoid-individual-operations.java
var items = (List<Map<String, Object>>) input.get("items");
var results = new ArrayList<>();
// Creating a step for each item - too many checkpoints!
for (int i = 0; i < items.size(); i++) {
    final var item = items.get(i);
    var result = ctx.step("process_item_" + i, Object.class,
        stepCtx -> processItem(item));
    results.add(result);
}
return results;
